05-networks/index.ts

```diff
@@ -0,0 +1,43 @@
+import * as http from "http";
+import * as url from "url";
+import { StringDecoder } from "string_decoder";
+import router from "./user/routes";
+import { IncomingMessageWithBody } from "./models/inccoming-message-with-body.interface";
+
+const server = http.createServer(
+  (req: IncomingMessageWithBody, res: http.ServerResponse) => {
+    const parsedUrl = url.parse(req.url!, true);
+    const paths: string[] = parsedUrl.pathname!.split("/").filter(Boolean);
+    const method = req.method!.toUpperCase();
+
+    const idIndex = paths.findIndex((item) => !isNaN(Number(item)));
+    if (idIndex >= 0) {
+      req.params = { id: paths[idIndex] };
+      paths[idIndex] = ":id";
+    }
+
+    const routeKey = `${paths.join("/")}:${method}`;
```

Comment:
It is an interesting approach to construct a route key this way and match it against existing routes.